ath79: Add support for Comfast E314N-v2

grosjo opened the pull request:
Taken code from https://patchwork.ozlabs.org/patch/884850/ that was never pushed by the author, and adapted to ath79.

The Comfast E314N-V2 is a 2.4 GHz 2x2 radio with a built-in directional antenna and a second Ethernet port - very similar to the Ubiquiti NanoStation M2. The Ethernet port features a pass-through PoE capability, enabled or disabled with a slide switch. The radio is built using a Qualcomm/Atheros QCA9531 chipset.

Firmware can be flashed on these units by the following method:
1.) Apply power to the unit
2.) Immediately AFTER applying power, hold down the reset button
3.) The WAN, LAN, and wireless lights will flash - wait three seconds (three flashes) and then release the button.
4.) After a second, the lights will flutter quickly and the unit will be visible at 192.168.1.1. A web page will be available to enable quick and simple uploading and flashing of firmware.

During the boot process, these units also look for a tftp server at 192.168.1.10. If one is present, the firmware can be uploaded as a file called firmware-auto.bin

Signed-off-by: Joan Moreau jom@grosjo.net

grosjo committed:
ath79: Add support for Comfast E314N-v2

Taken code from https://patchwork.ozlabs.org/patch/884850/ that was never pushed by the author, and adapted to ath79.

The Comfast E314N-V2 is a 2.4 GHz 2x2 radio with a built-in directional antenna and a second Ethernet port - very similar to the Ubiquiti NanoStation M2. The Ethernet port features a pass-through PoE capability, enabled or disabled with a slide switch.

Specifications :
- System-On-Chip: Qualcomm/Atheros QCA9531
- CPU/Speed: 650 MHz
- Flash size: 8 MiB
- RAM: 64 MiB
- 2 Ethernet 1Gbp
- 1 reset button
- 1 switch to choose PoE from LAN or Wan. 48Vdc
- Wifi 2.4 Ghz (b/g/n)
- UART inside the box (3.3V, pins marked on the PCB)

Firmware can be flashed on these units by the following method:
1.) Apply power to the unit
2.) Immediately AFTER applying power, hold down the reset button
3.) The WAN, LAN, and wireless lights will flash - wait three seconds (three flashes) and then release the button.
4.) After a second, the lights will flutter quickly and the unit will be visible at 192.168.1.1. A web page will be available to enable quick and simple uploading and flashing of firmware.

During the boot process, these units also look for a tftp server at 192.168.1.10. If one is present, the firmware can be uploaded as a file called firmware-auto.bin

Signed-off-by: Joan Moreau <jom@grosjo.net>

grosjo commented:
@ynezz Can you please kindly push this into master, so I do not need to redo thousands of times the same thing on github at each update ?

ynezz commented:
You're making it very hard for me to follow the changes, since you close and open the PR for almost every change. If I remember it correctly, I've asked you to attach bootlog/dmesg as well.

grosjo commented:
@fall513 Can you please check the changes ?

grosjo commented:
> You're making it very hard for me to follow the changes, since you close and open the PR for almost every change. If I remember it correctly, I've asked you to attach bootlog/dmesg as well.

@ynezz Sorry, github is making my life crazy. Hopefully this is stable

grosjo commented:
This is the boot log

However, same issue as in PR https://github.com/openwrt/openwrt/pull/1893 : No IP given to the device in recent git master

```
U-Boot 1.1.4-ga1cd7d4b-dirty (May 31 2018 - 16:44:04)

ap143 - Honey Bee 2.0

DRAM:  64 MB
Flash: 16 MB                                                                       
                                                                                   
dup 1 speed 100                                                                    
Using eth0 device                                                                  
ping failed;                                                                       
Hit any key to stop autoboot:  0 
## Booting image at 9f020000 ...
   Image Name:   MIPS OpenWrt Linux-4.14.109
   Created:      2019-04-20  23:27:54 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    1400636 Bytes =  1.3 MB
   Load Address: 80060000
   Entry Point:  80060000
   Verifying Checksum at 0x9f020040 ...OK
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 80060000) ...
## Giving linux memsize in bytes, 67108864

Starting kernel ...

[    0.000000] Linux version 4.14.109 (admin@gjserver) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9754-9fdb2c0ffd)) #0 Sat Apr 20 23:27:549
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
[    0.000000] MIPS: machine is COMFAST CF-E314N v2
[    0.000000] SoC: Qualcomm Atheros QCA9533 ver 2 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] random: get_random_bytes called from 0x8046573c with crng_init=0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 59004K/65536K available (3480K kernel code, 143K rwdata, 484K rodata, 1196K init, 197K bss, 6532K reserved, 0K )
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 650.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5880801374 ns
[    0.000010] sched_clock: 32 bits at 325MHz, resolution 3ns, wraps every 6607641598ns
[    0.008265] Calibrating delay loop... 432.53 BogoMIPS (lpj=2162688)
[    0.074881] pid_max: default: 32768 minimum: 301
[    0.079963] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086938] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.099012] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.109458] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.115996] pinctrl core: initialized pinctrl subsystem
[    0.122716] NET: Registered protocol family 16
[    0.153513] clocksource: Switched to clocksource MIPS
[    0.160184] NET: Registered protocol family 2
[    0.165719] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.173090] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.179875] TCP: Hash tables configured (established 1024 bind 1024)
[    0.186795] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.192978] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.199971] NET: Registered protocol family 1
[    0.210244] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.222719] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.228932] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.251260] io scheduler noop registered
[    0.255477] io scheduler deadline registered (default)
[    0.263351] pinctrl-single 1804002c.pinmux: 576 pins at pa b804002c size 72
[    0.271884] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.279510] console [ttyS0] disabled
[    0.283334] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 8, base_baud = 1562500) is a 16550A
[    0.292504] console [ttyS0] enabled
[    0.292504] console [ttyS0] enabled
[    0.300073] bootconsole [early0] disabled
[    0.300073] bootconsole [early0] disabled
[    0.316308] m25p80 spi0.0: found en25qh64, expected w25q128
[    0.322099] m25p80 spi0.0: en25qh64 (8192 Kbytes)
[    0.327088] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.333668] Creating 4 MTD partitions on "spi0.0":
[    0.338639] 0x000000000000-0x000000010000 : "u-boot"
[    0.344798] 0x000000010000-0x000000020000 : "art"
[    0.350552] 0x000000020000-0x000000ff0000 : "firmware"
[    0.355957] mtd: partition "firmware" extends beyond the end of device "spi0.0" -- size truncated to 0x7e0000
[    0.370340] 2 uimage-fw partitions found on MTD device firmware
[    0.376540] Creating 2 MTD partitions on "firmware":
[    0.381677] 0x000000000000-0x000000160000 : "kernel"
[    0.387701] 0x000000160000-0x0000007e0000 : "rootfs"
[    0.393750] mtd: device 4 (rootfs) set to be root filesystem
[    0.399642] 1 squashfs-split partitions found on MTD device rootfs
[    0.406065] 0x0000003d0000-0x0000007e0000 : "rootfs_data"
[    0.412505] 0x000000ff0000-0x000001000000 : "art-backup"
[    0.418066] mtd: partition "art-backup" is out of reach -- disabled
[    0.426775] libphy: Fixed MDIO Bus: probed
[    0.773962] libphy: ag71xx_mdio: probed
[    0.779389] libphy: ar8xxx-mdio: probed
[    0.792582] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.185861] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.196483] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    1.203074] ag71xx 1a000000.eth: invalid MAC address, using random address
[    1.545425] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.555465] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.563348] leds-gpio leds: Led cf-e314n-v2:white:signal1 renamed to cf-e314n-v2:white:signal1_1 due to name collision
[    1.575287] NET: Registered protocol family 17
[    1.579974] 8021q: 802.1Q VLAN Support v1.8
[    1.595214] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.612296] Freeing unused kernel memory: 1196K
[    1.617012] This architecture does not have kernel memory protection.
[    2.266360] init: Console is alive
[    2.270163] init: - watchdog -
[    2.498411] random: fast init done
[    3.224476] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.311211] usbcore: registered new interface driver usbfs
[    3.317044] usbcore: registered new interface driver hub
[    3.322666] usbcore: registered new device driver usb
[    3.334754] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    3.343635] ehci-platform: EHCI generic platform driver
[    3.352073] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    3.360125] ohci-platform: OHCI generic platform driver
[    3.366832] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.384962] init: - preinit -
[    4.407002] random: jshn: uninitialized urandom read (4 bytes read)
[    4.633758] random: jshn: uninitialized urandom read (4 bytes read)
[    4.702611] random: jshn: uninitialized urandom read (4 bytes read)
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    7.056713] eth0: link up (100Mbps/Full duplex)
[    8.254589] jffs2: notice: (430) jffs2_build_xattr_subsystem: complete building xattr subsystem, 6 of xdatum (3 unchecked, 3 orphan.
[    8.272221] mount_root: switching to jffs2 overlay
[    8.304233] overlayfs: upper fs does not support tmpfile.
[    8.315894] urandom-seed: Seeding with /etc/urandom.seed
[    8.498819] eth0: link down
[    8.526294] procd: - early -
[    8.529415] procd: - watchdog -
[    9.166732] procd: - watchdog -
[    9.170340] procd: - ubus -
[    9.244890] urandom_read: 5 callbacks suppressed
[    9.244900] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.259588] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.267882] procd: - init -
Please press Enter to activate this console.
[    9.732414] kmodloader: loading kernel modules from /etc/modules.d/*
[    9.744839] Loading modules backported from Linux version v4.19.23-0-g67d52fae61c1
[    9.752667] Backport generated by backports.git v4.19.23-1-0-g480a925a
[    9.767706] nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
[    9.845680] xt_time: kernel timezone is -0000
[    9.873791] ip_tables: (C) 2000-2006 Netfilter Core Team
[    9.926829] PPP generic driver version 2.4.2
[    9.934294] NET: Registered protocol family 24
[   10.010772] ieee80211 phy0: Atheros AR9531 Rev:2 mem=0xb8100000, irq=11
[   10.054069] kmodloader: done loading kernel modules from /etc/modules.d/*
[   21.276380] br-lan: port 1(eth1) entered blocking state
[   21.281799] br-lan: port 1(eth1) entered disabled state
[   21.287652] device eth1 entered promiscuous mode
[   21.384042] br-wan: port 1(eth0) entered blocking state
[   21.389463] br-wan: port 1(eth0) entered disabled state
[   21.395289] device eth0 entered promiscuous mode
[   22.335456] eth1: link up (1000Mbps/Full duplex)
[   22.340285] br-lan: port 1(eth1) entered blocking state
[   22.345763] br-lan: port 1(eth1) entered forwarding state
[   23.548438] eth0: link up (100Mbps/Full duplex)
[   23.553189] br-wan: port 1(eth0) entered blocking state
[   23.558648] br-wan: port 1(eth0) entered forwarding state
[  130.903565] random: crng init done

```

grosjo commented:
yeehe, added i2c module, now working very fine:

[    0.000000] Linux version 4.14.113 (admin@gjserver) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9915-d6643aca34)) #0 Wed May 1 05:25:55 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
[    0.000000] MIPS: machine is COMFAST CF-E314N v2
[    0.000000] SoC: Qualcomm Atheros QCA9533 ver 2 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000] free_area_init_node: node 0, pgdat 80455e90, node_mem_map 81007ce0
[    0.000000]   Normal zone: 128 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from 0x8045d73c with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 59004K/65536K available (3490K kernel code, 161K rwdata, 424K rodata, 1228K init, 197K bss, 6532K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 650.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5880801374 ns
[    0.000011] sched_clock: 32 bits at 325MHz, resolution 3ns, wraps every 6607641598ns
[    0.008270] Calibrating delay loop... 432.53 BogoMIPS (lpj=2162688)
[    0.074882] pid_max: default: 32768 minimum: 301
[    0.079971] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086946] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.100140] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.110593] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.117129] pinctrl core: initialized pinctrl subsystem
[    0.123871] NET: Registered protocol family 16
[    0.131890] PCI host bridge /ahb/apb/pcie-controller@180c0000 ranges:
[    0.138774]  MEM 0x0000000010000000..0x0000000013ffffff
[    0.144275]   IO 0x0000000000000000..0x0000000000000000
[    0.149796] ar724x-pci 180c0000.pcie-controller: PCIe link is down
[    0.177678] PCI host bridge to bus 0000:00
[    0.182095] pci_bus 0000:00: root bus resource [mem 0x10000000-0x13ffffff]
[    0.189344] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.195234] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.202391] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.210784] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.213903] clocksource: Switched to clocksource MIPS
[    0.220652] NET: Registered protocol family 2
[    0.226217] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.233582] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.240366] TCP: Hash tables configured (established 1024 bind 1024)
[    0.247281] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.253463] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.260463] NET: Registered protocol family 1
[    0.265138] PCI: CLS 0 bytes, default 32
[    0.270468] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.283201] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.289426] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.311770] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.319655] io scheduler noop registered
[    0.323782] io scheduler deadline registered (default)
[    0.331635] pinctrl-single 1804002c.pinmux: 576 pins at pa b804002c size 72
[    0.340225] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.347879] console [ttyS0] disabled
[    0.351700] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 1562500) is a 16550A
[    0.360882] console [ttyS0] enabled
[    0.368448] bootconsole [early0] disabled
[    0.385565] m25p80 spi0.0: en25qh64 (8192 Kbytes)
[    0.390494] 5 fixed-partitions partitions found on MTD device spi0.0
[    0.397127] Creating 5 MTD partitions on "spi0.0":
[    0.402090] 0x000000000000-0x000000010000 : "u-boot"
[    0.408248] 0x000000010000-0x000000020000 : "art"
[    0.414073] 0x000000020000-0x0000007e0000 : "firmware"
[    0.422843] 2 uimage-fw partitions found on MTD device firmware
[    0.429045] Creating 2 MTD partitions on "firmware":
[    0.434196] 0x000000000000-0x000000160000 : "kernel"
[    0.440240] 0x000000160000-0x0000007c0000 : "rootfs"
[    0.446263] mtd: device 4 (rootfs) set to be root filesystem
[    0.452154] 1 squashfs-split partitions found on MTD device rootfs
[    0.458623] 0x0000003a0000-0x0000007c0000 : "rootfs_data"
[    0.465068] 0x0000007e0000-0x0000007f0000 : "configs"
[    0.471108] 0x0000007f0000-0x000000800000 : "nvram"
[    0.478524] libphy: Fixed MDIO Bus: probed
[    0.824377] libphy: ag71xx_mdio: probed
[    0.829808] libphy: ar8xxx-mdio: probed
[    0.843042] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.235875] libphy: ag71xx_mdio: probed
[    1.241097] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.251680] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    1.595824] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.605894] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.614647] NET: Registered protocol family 17
[    1.619328] 8021q: 802.1Q VLAN Support v1.8
[    1.634114] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.651462] Freeing unused kernel memory: 1228K
[    1.656185] This architecture does not have kernel memory protection.
[    2.295698] init: Console is alive
[    2.299496] init: - watchdog -
[    2.516289] random: fast init done
[    3.186365] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.304307] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.322374] init: - preinit -
[    4.275654] random: jshn: uninitialized urandom read (4 bytes read)
[    4.369801] random: jshn: uninitialized urandom read (4 bytes read)
[    4.438185] random: jshn: uninitialized urandom read (4 bytes read)
[    6.847170] eth0: link up (100Mbps/Full duplex)
[    8.089817] jffs2: notice: (429) jffs2_build_xattr_subsystem: complete building xattr subsystem, 3 of xdatum (0 unchecked, 1 orphan) and 5 of xref (1 dead, 0 orphan) found.
[    8.107435] mount_root: switching to jffs2 overlay
[    8.141298] overlayfs: upper fs does not support tmpfile.
[    8.152687] urandom-seed: Seeding with /etc/urandom.seed
[    8.197937] eth0: link down
[    8.215530] procd: - early -
[    8.218645] procd: - watchdog -
[    8.852453] procd: - watchdog -
[    8.856136] procd: - ubus -
[    8.992018] urandom_read: 5 callbacks suppressed
[    8.992028] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.023598] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.045824] procd: - init -
[    9.465398] kmodloader: loading kernel modules from /etc/modules.d/*
[    9.478369] i2c /dev entries driver
[    9.494456] Loading modules backported from Linux version v4.19.32-0-g3a2156c839c7
[    9.502288] Backport generated by backports.git v4.19.32-1-0-g1c4f7569
[    9.517423] nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
[    9.616121] xt_time: kernel timezone is -0000
[    9.646062] ip_tables: (C) 2000-2006 Netfilter Core Team
[    9.701154] PPP generic driver version 2.4.2
[    9.708686] NET: Registered protocol family 24
[    9.764913] ath: EEPROM regdomain: 0x0
[    9.764926] ath: EEPROM indicates default country code should be used
[    9.764930] ath: doing EEPROM country->regdmn map search
[    9.764949] ath: country maps to regdmn code: 0x3a
[    9.764955] ath: Country alpha2 being used: US
[    9.764959] ath: Regpair used: 0x3a
[    9.779484] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[    9.781352] ieee80211 phy0: Atheros AR9531 Rev:2 mem=0xb8100000, irq=13
[    9.868974] kmodloader: done loading kernel modules from /etc/modules.d/*
[   19.314890] br-lan: port 1(eth0) entered blocking state
[   19.320302] br-lan: port 1(eth0) entered disabled state
[   19.326095] device eth0 entered promiscuous mode
[   20.480955] eth1: link up (1000Mbps/Full duplex)
[   21.397106] eth0: link up (100Mbps/Full duplex)
[   21.401870] br-lan: port 1(eth0) entered blocking state
[   21.407334] br-lan: port 1(eth0) entered forwarding state

grosjo commented:
@ynezz is that ok for you ?

ynezz commented:
> is that ok for you ?

No, content of the dmesg you've provided doesn't correspond with the DTS, please update your dmesg, thanks.

grosjo commented:
Well, that is actually what I have from my device, running on compiled git master with this patch. I can not invent it !
What do you /think/ is wrong ?

grosjo commented:
@ynezz , one more time, the dmesg (the same, but based on latest git master) :

```
root@OpenWrt:~# dmesg
[    0.000000] Linux version 4.14.115 (admin@gjserver) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9945-bc85640cdc)) #0 Mon May 6 10:43:01 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
[    0.000000] MIPS: machine is COMFAST CF-E314N v2
[    0.000000] SoC: Qualcomm Atheros QCA9533 ver 2 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000] free_area_init_node: node 0, pgdat 80455e90, node_mem_map 81007ce0
[    0.000000]   Normal zone: 128 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from 0x8045d73c with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 59004K/65536K available (3491K kernel code, 160K rwdata, 424K rodata, 1228K init, 197K bss, 6532K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 650.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5880801374 ns
[    0.000010] sched_clock: 32 bits at 325MHz, resolution 3ns, wraps every 6607641598ns
[    0.008268] Calibrating delay loop... 432.53 BogoMIPS (lpj=2162688)
[    0.074881] pid_max: default: 32768 minimum: 301
[    0.079966] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086938] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.100102] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.110550] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.117088] pinctrl core: initialized pinctrl subsystem
[    0.123838] NET: Registered protocol family 16
[    0.131776] PCI host bridge /ahb/apb/pcie-controller@180c0000 ranges:
[    0.138663]  MEM 0x0000000010000000..0x0000000013ffffff
[    0.144167]   IO 0x0000000000000000..0x0000000000000000
[    0.149683] ar724x-pci 180c0000.pcie-controller: PCIe link is down
[    0.177141] PCI host bridge to bus 0000:00
[    0.181553] pci_bus 0000:00: root bus resource [mem 0x10000000-0x13ffffff]
[    0.188794] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.194676] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.201825] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.210211] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.213295] clocksource: Switched to clocksource MIPS
[    0.220046] NET: Registered protocol family 2
[    0.225606] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.232976] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.239767] TCP: Hash tables configured (established 1024 bind 1024)
[    0.246681] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.252865] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.259869] NET: Registered protocol family 1
[    0.264549] PCI: CLS 0 bytes, default 32
[    0.269824] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.282462] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.288684] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.310915] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 252)
[    0.318795] io scheduler noop registered
[    0.322917] io scheduler deadline registered (default)
[    0.330756] pinctrl-single 1804002c.pinmux: 576 pins at pa b804002c size 72
[    0.339342] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.346989] console [ttyS0] disabled
[    0.350810] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 1562500) is a 16550A
[    0.359988] console [ttyS0] enabled
[    0.367550] bootconsole [early0] disabled
[    0.383888] m25p80 spi0.0: en25qh64 (8192 Kbytes)
[    0.388825] 5 fixed-partitions partitions found on MTD device spi0.0
[    0.395440] Creating 5 MTD partitions on "spi0.0":
[    0.400401] 0x000000000000-0x000000010000 : "u-boot"
[    0.406541] 0x000000010000-0x000000020000 : "art"
[    0.412246] 0x000000020000-0x0000007e0000 : "firmware"
[    0.421672] 2 uimage-fw partitions found on MTD device firmware
[    0.427875] Creating 2 MTD partitions on "firmware":
[    0.433012] 0x000000000000-0x000000160000 : "kernel"
[    0.439080] 0x000000160000-0x0000007c0000 : "rootfs"
[    0.445116] mtd: device 4 (rootfs) set to be root filesystem
[    0.451013] 1 squashfs-split partitions found on MTD device rootfs
[    0.457476] 0x0000003a0000-0x0000007c0000 : "rootfs_data"
[    0.463994] 0x0000007e0000-0x0000007f0000 : "configs"
[    0.470057] 0x0000007f0000-0x000000800000 : "nvram"
[    0.477492] libphy: Fixed MDIO Bus: probed
[    0.823766] libphy: ag71xx_mdio: probed
[    0.829193] libphy: ar8xxx-mdio: probed
[    0.842370] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.235251] libphy: ag71xx_mdio: probed
[    1.240463] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    1.251051] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    1.595205] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.605274] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.614019] NET: Registered protocol family 17
[    1.618706] 8021q: 802.1Q VLAN Support v1.8
[    1.633381] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.650732] Freeing unused kernel memory: 1228K
[    1.655443] This architecture does not have kernel memory protection.
[    2.294814] init: Console is alive
[    2.298625] init: - watchdog -
[    2.515387] random: fast init done
[    3.185361] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.303083] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.321199] init: - preinit -
[    4.275468] random: jshn: uninitialized urandom read (4 bytes read)
[    4.369099] random: jshn: uninitialized urandom read (4 bytes read)
[    4.438213] random: jshn: uninitialized urandom read (4 bytes read)
[    6.850182] eth0: link up (100Mbps/Full duplex)
[    8.089624] jffs2_scan_eraseblock(): End of filesystem marker found at 0x10000
[    8.097137] jffs2_build_filesystem(): unlocking the mtd device... 
[    8.097143] done.
[    8.105508] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   22.455174] done.
[   22.464972] jffs2: notice: (430) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   22.482482] mount_root: overlay filesystem has not been fully initialized yet
[   22.494376] mount_root: switching to jffs2 overlay
[   22.526592] overlayfs: upper fs does not support tmpfile.
[   22.790965] urandom-seed: Seed file not found (/etc/urandom.seed)
[   22.827425] eth0: link down
[   22.844704] procd: - early -
[   22.847819] procd: - watchdog -
[   23.481220] procd: - watchdog -
[   23.484907] procd: - ubus -
[   23.619837] urandom_read: 5 callbacks suppressed
[   23.619847] random: ubusd: uninitialized urandom read (4 bytes read)
[   23.650928] random: ubusd: uninitialized urandom read (4 bytes read)
[   23.659107] procd: - init -
[   24.089324] kmodloader: loading kernel modules from /etc/modules.d/*
[   24.101919] i2c /dev entries driver
[   24.118018] Loading modules backported from Linux version v4.19.32-0-g3a2156c839c7
[   24.125925] Backport generated by backports.git v4.19.32-1-0-g1c4f7569
[   24.140827] nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
[   24.218714] xt_time: kernel timezone is -0000
[   24.247228] ip_tables: (C) 2000-2006 Netfilter Core Team
[   24.299144] PPP generic driver version 2.4.2
[   24.306688] NET: Registered protocol family 24
[   24.365812] ath: EEPROM regdomain: 0x0
[   24.365825] ath: EEPROM indicates default country code should be used
[   24.365828] ath: doing EEPROM country->regdmn map search
[   24.365847] ath: country maps to regdmn code: 0x3a
[   24.365854] ath: Country alpha2 being used: US
[   24.365857] ath: Regpair used: 0x3a
[   24.380446] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   24.382311] ieee80211 phy0: Atheros AR9531 Rev:2 mem=0xb8100000, irq=13
[   24.490774] kmodloader: done loading kernel modules from /etc/modules.d/*
[   34.726428] br-lan: port 1(eth0) entered blocking state
[   34.731847] br-lan: port 1(eth0) entered disabled state
[   34.737643] device eth0 entered promiscuous mode
[   35.925167] eth1: link up (1000Mbps/Full duplex)
[   36.836474] eth0: link up (100Mbps/Full duplex)
[   36.841231] br-lan: port 1(eth0) entered blocking state
[   36.846699] br-lan: port 1(eth0) entered forwarding state
```

grosjo commented:
@ynezz is that ok for you ?

grosjo commented:
@ynezz what woudl be the next step ?

adrianschmutzler commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,165 @@
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
```
You may try using gpio-keys and removing poll-interval here.

grosjo replied:
WHat is the difference ?
and what would be the correction ?

adrianschmutzler replied:
https://github.com/openwrt/openwrt/pull/2056#discussion_r285337716

```
compatible = "gpio-keys"
```

grosjo replied:
done

adrianschmutzler commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,165 @@
+			};
+
+			partition@7e0000 {
+				label = "configs";
```
configs => config ?

grosjo replied:
this is the name as per vendor

adrianschmutzler replied:
Then it's okay, just a question.

adrianschmutzler commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,165 @@
+				reg = <0x7e0000 0x010000>;
+			};
+	
+			partition@7f000 {
```
Must be 7f0000 instead of 7f000 !

grosjo commented:
@adrianschmutzler @ynezz How to get that included in 19.07 ?

adrianschmutzler commented:
@grosjo I'm not a dev here, but normally device support PRs are not backported. However, I do not know whether this is also true in this early phase before first RC.

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -171,6 +171,14 @@ define Device/comfast_cf-e120a-v3
 
+define Device/comfast_cf-e314n-v2
+  ATH_SOC := qca9531
+  DEVICE_TITLE := COMFAST CF-E314N v2
```
Please update to use DEVICE_VENDOR, DEVICE_MODEL and DEVICE_VARIANT as introduced in:
https://github.com/openwrt/openwrt/commit/8c3dbaf8b49233e275b2dce4e30dc0a2bf8c86c5

grosjo replied:
@adrianschmutzler Problem with such setup, is that it does not appear in the "make menuconfig" setup

adrianschmutzler replied:
Sounds like you haven't rebased onto recent master?

grosjo replied:
I tested against 19.07
INdeed it works with master, but main branch is 19.07 right ?

adrianschmutzler replied:
You should always develop against master.
After 19.07 has been made it's own branch, only bugfixes will be merged there (= stable branch).
Device support patches are (almost) never backported. If you desire to have this device in 19.07, you will have to provide a backport patch in your local repository.

adrianschmutzler commented:
I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

adrianschmutzler commented:
Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with `cat`, but have to use the following code to read from the paths retrieved by `find`:
```
. /lib/functions/system.sh
get_mac_binary "$yourpath" 0
```
The include is obviously only required once. Sorry.

grosjo commented:
@adrianschmutzler  maybe I can answer next month

@ynezz  How to get this pushed to master ?

adrianschmutzler commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+&wmac {
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+};
```
Is the MAC address on wmac set correctly with this or is there a mtd-mac-address missing?

grosjo replied:
all this works very fine

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -171,6 +171,16 @@ define Device/comfast_cf-e120a-v3
+define Device/comfast_cf-e314n-v2
+  ATH_SOC := qca9531
+  DEVICE_VENDOR := COMFAST
+  DEVICE_MODEL := E314N
```
This should be "CF-E314N".

grosjo replied:
done

grosjo commented:
@adrianschmutzler
/proc/device-tree/ahb/eth@1a000000/mac-address   <== this one (LAN)
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address   <== this one (WAN)
/proc/device-tree/ahb/eth@19000000/mtd-mac-address

adrianschmutzler commented:
> @adrianschmutzler
> /proc/device-tree/ahb/eth@1a000000/mac-address <== this one (LAN)
> /proc/device-tree/ahb/eth@1a000000/mtd-mac-address
> /proc/device-tree/ahb/eth@19000000/mac-address <== this one (WAN)
> /proc/device-tree/ahb/eth@19000000/mtd-mac-address

Thanks. Which of those is on the device's label?

grosjo commented:
> Thanks. Which of those is on the device's label?
No label

grosjo commented:
@ynezz how to get that pushed to master ? Thank you

grosjo commented:
@blocktrron Can you review this as well ? That device is really common in the market and it would be great to have it in master

grosjo commented:
@blocktrron Can you review this as well ? That device is really common in the market and it would be great to have it in master

grosjo commented:
@blocktrron Can you review this as well ? That device is really common in the market and it would be great to have it in master

blocktrron commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -47,6 +47,10 @@ comfast,cf-e120a-v3)
 	ucidef_set_led_rssi "rssimediumhigh" "RSSIMEDIUMHIGH" "$boardname:green:rssimediumhigh" "wlan0" "51" "100"
 	ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "$boardname:green:rssihigh" "wlan0" "76" "100"
 	;;
+comfast,cf-e314n-v2)
```
RSSILEDs seem to be missing here.

grosjo replied:
done

grosjo replied:
done

blocktrron commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+		serial0 = &uart;
+		led-boot = &wan;
+		led-failsafe = &wan;
+		led-upgrade = &wan;
```
If there's no LED to display SYS-activity, we do not have to come up with one. So rather drop this than to misuse the WAN led.

adrianschmutzler replied:
For the TP-Link CPE devices we consistently used RSSI LED 4 (max) for that:
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9533_tplink_cpe210.dtsi#L44

In contrast to LAN/WAN, a single RSSI led blinking cannot be mistaken so easily.

blocktrron replied:
I'm aware of that (and i don't like this more).

Assigning a LED to multiple functions makes them ambiguos.
For WLAN / LAN LEDs, i see this is a big nono, as e.g. failsafe can't be displayed clearly.

However, if we already have this ambiguity there, you can also introduce it here.

grosjo replied:
actually, there is a "system" led on the board. Not sure how to figure out the GPIO number thought

grosjo replied:
actually, there is a "system" led on the board. Not sure how to figure out the GPIO number thought

adrianschmutzler replied:
It might also happen that you cannot access the LED at all. That's the case for the CPE devices mentioned earlier, they have a system/power which only can be on...

grosjo replied:
so we keep it as is ?

grosjo replied:
so we keep it as is ?

adrianschmutzler replied:
If I understood @blocktrron correctly, you can either use rssi led max as we did for the TP-Link CPE devices or you can use no system led at all (and remove the aliases). But you shouldn't keep WAN.

blocktrron commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+
+			partition@7e0000 {
+				label = "configs";
+				reg = <0x7e0000 0x010000>;
```
read-only flag is missing here.

grosjo replied:
done

blocktrron commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+	
+			partition@7f0000 {
+				label = "nvram";
+				reg = <0x7f0000 0x010000>;
```
read-only flag is missing here.

grosjo replied:
done

blocktrron commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+	mtd-cal-data = <&ART 0x1000>;
+};
+
+&gpio {
```
`gpio` property should be save to drop, as it's not disabled in qca9557.dtsi. Can you give it a shot?

grosjo replied:
just tried it. Works fine without gpio paragraph

grosjo replied:
just tried it. Works fine without gpio paragraph

grosjo replied:
just tried it. Works fine without gpio paragraph

blocktrron commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,164 @@
+				read-only;
+			};
+
+			ART: partition@10000 {
```
`ART` phandle in lowercase please.

grosjo replied:
done

blocktrron commented on target/linux/ath79/image/generic.mk:
```diff
@@ -171,6 +171,15 @@ define Device/comfast_cf-e120a-v3
+  DEVICE_VENDOR := COMFAST
+  DEVICE_MODEL := CF-E314N
+  DEVICE_VARIANT := v2
+  DEVICE_PACKAGES := rssileds kmod-leds-gpio kmod-leds-reset kmod-i2c-gpio
```
`kmod-leds-gpio kmod-leds-reset kmod-i2c-gpio` should all not be necessary here.

grosjo replied:
but kmod-leds-gpio yes, right ?

blocktrron replied:
It's compiled into the kernel, so you do not need to add the module. See https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/config-4.14#L134

grosjo replied:
ok done

blocktrron reviewed (commented):
Thanks for your PR (again :smile: )

See my suggestions below.

blocktrron commented:
Checkpatch found the following:

```
WARNING: Possible unwrapped commit description (prefer a maximum 75 chars per line)
#6: 
Taken code from https://patchwork.ozlabs.org/patch/884850/ that was never pushed by the author, and adapted to ath79.

WARNING: 'swith' may be misspelled - perhaps 'switch'?
#17: 
- 1 swith to choose PoE from LAN or Wan. 48Vdc

ERROR: Unrecognized email address: 'Joan Moreau jom@grosjo.net'
#29: 
Signed-off-by: Joan Moreau jom@grosjo.net

ERROR: trailing whitespace
#188: FILE: target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:126:
+^I$

WARNING: Missing Signed-off-by: line by nominal patch author 'Joan Moreau <jom@grosjo.net>'
```

Also - could you add details about the UART port (pinout / connection-parameters)?

grosjo commented:
Another issue : A previous commit echange eth0 and eth1
https://github.com/openwrt/openwrt/pull/2066

This impact wrongly this device. How to counter effect the previous patch ?

adrianschmutzler commented:
You will have to live with the eth0/eth1 swap.

grosjo commented:
Not . The port are switched since PR 2066 (i.e. WAN is connected to LAN and vice versa). How can I fix that ?

grosjo commented:
trying to use 02_network

grosjo commented:
trying to use 02_network

adrianschmutzler commented:
Ah, now I get it: If you want to change what's lan/wan, you indeed need 02_network. What you can't change (easily) is the eth0/eth1 order.

grosjo commented:
Success ! All works fine now, including WAN to wan port and LAN to lan port and according leds.
I fixed also the bits and bolts you mentionned earlier
@blocktrron  The ball is in your hands now

grosjo commented:
@blocktrron Ready to merge ?

adrianschmutzler commented on target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts:
```diff
@@ -0,0 +1,155 @@
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+};
+
```
There is an empty line here. But this could also be done during merge ...

grosjo commented:
I removed the space for you 

Can we now push this to master ?

On 2019-08-09 05:51, Adrian Schmutzler wrote:

> @adrianschmutzler commented on this pull request. 
> -------------------------
> 
> In target/linux/ath79/dts/qca9531_comfast_cf-e314n-v2.dts [1]:
> 
>> +&eth0 {
> +    status = "okay";
> +    mtd-mac-address = <&art 0x0>;
> +    phy-handle = <&swphy4>;
> +};
> +
> +&eth1 {
> +    status = "okay";
> +    mtd-mac-address = <&art 0x6>;
> +};
> +
> +&wmac {
> +    status = "okay";
> +    mtd-cal-data = <&art 0x1000>;
> +};
> +
> 
> There is an empty line here. But this could also be done during merge ... 
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [2], or mute the thread [3].
 

Links:
------
[1] https://github.com/openwrt/openwrt/pull/1955#discussion_r312258289
[2]
https://github.com/openwrt/openwrt/pull/1955?email_source=notifications&amp;email_token=AAROMB4IFJAK64GYVDXNTMTQDSINBA5CNFSM4HCO6SV2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCBBXPEI#pullrequestreview-272856977
[3]
https://github.com/notifications/unsubscribe-auth/AAROMBZV2PIWAM6YNAFXYELQDSINBANCNFSM4HCO6SVQ

blocktrron commented:
Thanks for your Pull Request! I've merged your patch into my staging tree: https://git.openwrt.org/?p=openwrt/staging/blocktrron.git;a=summary

Note I've fixed two misformats in your commit message.
1. The first word after the target prefix shouldn't be capitalized
2. Your commit message must have a length of max. 75 chars per line. I've wrapped them for you. See also https://openwrt.org/submitting-patches

Neustradamus commented:
Merged commit:
- https://github.com/openwrt/openwrt/commit/abbbecaa73dcb5a36932a224f74488d0c5e3b3cd
